### ISSUE TITLE
Type instantiation is excessively deep and possibly infinite .ts(2589) when using Type.Exclude

### ISSUE THREAD

**Issue by espenja** [sinclairzx81/typebox#737]
The following simple example produces the TypeScript error message:
```
⚠ Error(TS2589) | 
Type instantiation is excessively deep and possibly infinite.
(alias) const Type: JavaScriptTypeBuilder
import Type
JavaScript Type Builder with Static Resolution for TypeScript
```

```typescript
import { type Static, Type } from "@sinclair/typebox"

export type TypeTest = Static<typeof TypeTest>
export const TypeTest = Type.Union([Type.Literal("array"), Type.Literal("text")])

export type FieldTest = Static<typeof FieldTest>
export const FieldTest = Type.Object({
	type: Type.Exclude(TypeTest, Type.Literal("text"))
})
```

**Comment by sinclairzx81** [comment]
@espenja Hi, thanks for reporting.

Yes, definitely a bug. Have pushed a fix for this on 0.31.11. This should resolve instantiation issues for both Exclude and Extract when embedded in outer types.

Cheers

**Comment by sinclairzx81** [comment]
@espenja Heya,

Will close off this issue for now as this should be resolved. But if you run into any issues, please feel free to reply on this thread.

All the best!
S
